chatterbot/trainers.py

```diff
@@ -78,17 +77,25 @@ def train(self, conversation):
         Train the chat bot based on the provided list of
         statements that represents a single conversation.
         """
-        previous_statement_text = None
 
-        for text in conversation:
-            statement = self.get_or_create(text)
+        # Inline method yields statements for larger datasets
+        def get_statements(conversations):
+            previous_statement_text = None
+
+            for text in conversation:
+                statement = self.get_or_create(text)
+
+                if previous_statement_text:
+                    statement.add_response(
+                        Response(previous_statement_text)
+                    )
+
+                previous_statement_text = statement.text
+                yield statement
```

Comment:
> Was there a performance increase after this change?

It seems to me both are same.

> These changes are quite different from what was discussed in #815.

I thought i will combine both pull requests into one, now i will remove yield statements, I will push some more commit what we have disucssed.

> Generator functions are a bit different from list comprehension as they produce a "lazy list" 

Any more information about ``lazy lists``